src/core/main.c

```diff
@@ -354,10 +354,13 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                 if (proc_cmdline_value_missing(key, value))
                         return 0;
 
-                r = id128_from_string_nonzero(value, &arg_machine_id);
-                if (r < 0)
-                        log_warning_errno(r, "MachineID '%s' is not valid, ignoring: %m", value);
-
+                if (streq(value, "firmware")) {
+                        arg_machine_id = SD_ID128_FIRMWARE;
```

Comment:
no, please don't. just add `static bool arg_machine_id_from_firmware = false;` and do not overload the existing arg_machine_id like this with yet another special case.